addons/product/models/product.py

```diff
@@ -0,0 +1,870 @@
+        context = self.env.context
+        if context.get('currency_id'):
+            currency = self.env.user.company_id.currency_id
+            to_cureency = self.env['res.currency'].browse(context['currency_id'])
+        res = {}
+        for product in products:
+            # standard_price field can only be seen by users in base.group_user
+            # Thus, in order to compute the sale price from the cost for users not in this group
+            # We fetch the standard price as the superuser
+            if ptype != 'standard_price':
+                res[product.id] = product[ptype] or 0.0
+            else:
+                product = product.with_context(force_company=product.env.user.company_id.id)
+                res[product.id] = res[product.id] = product.sudo()[ptype]
+            if ptype == 'list_price':
+                res[product.id] += product._name == "product.product" and float(product.price_extra) or 0.0
+            if 'uom' in context:
+                res[product.id] = product.uom_id._compute_price(res[product.id], context['uom'])
+            # Convert from current user company currency to asked one
+            if 'currency_id' in context:
+                # Take current user company currency.
+                # This is right cause a field cannot be in more than one currency
+                res[product.id] = currency.compute(res[product.id], to_cureency)
+        return res
+
+    @api.onchange('uom_id', 'uom_po_id')
+    def onchange_uom(self):
+        if self.uom_id:
+            self.uom_po_id = self.uom_id
+
+    @api.multi
+    def create_variant_ids(self):
+        ctx = self.env.context.copy()
+        Product = self.env["product.product"]
+        if ctx.get("create_product_variant"):
+            return None
+
+        ctx.update(active_test=False, create_product_variant=True)
+
+        for tmpl_id in self.with_context(ctx):
+
+            # list of values combination
+            variant_alone = []
+            all_variants = [[]]
+            for variant_id in tmpl_id.attribute_line_ids:
+                if len(variant_id.value_ids) == 1:
+                    variant_alone.append(variant_id.value_ids[0])
+                temp_variants = []
+                for variant in all_variants:
+                    for value_id in variant_id.value_ids:
+                        temp_variants.append(sorted(variant + [int(value_id)]))
+                if temp_variants:
+                    all_variants = temp_variants
+
+            # adding an attribute with only one value should not recreate product
+            # write this attribute on every product to make sure we don't lose them
+            for variant_id in variant_alone:
+                product_ids = []
+                for product_id in tmpl_id.product_variant_ids:
+                    if variant_id.id not in map(int, product_id.attribute_value_ids):
+                        product_ids.append(product_id.id)
```

Comment:
Use set operation on recordset(s) instead of appending id(s) and browsing that id(s) to call the method.